java/client/src/main/java/glide/utils/ArrayTransformUtils.java

```diff
@@ -55,6 +55,15 @@ public static <T, U extends T> U[] castArray(T[] objectArr, Class<U> clazz) {
                 .toArray(size -> (U[]) Array.newInstance(clazz, size));
     }
 
+    public static <T, U extends T> Map<String, Map<String, U[]>> castMapOfMaps(
+            Map<String, Object> mapOfMapOfArrays, Class<U> clazz) {
```

Comment:
Probably, you don't need the last arg
probably, you don't need that function at all